src/org/mozilla/javascript/Parser.java

```diff
@@ -3812,6 +3793,25 @@ private StringLiteral createStringLiteral() {
         return s;
     }
 
+    private NumberLiteral createNumberLiteral() {
+        String s = ts.getString();
+        if (this.inUseStrictDirective && ts.isNumberOldOctal()) {
```

Comment:
This pull request is a change in strict mode.
It is a specification from ES5 to prohibit old octal numbers in strict mode.
https://www.ecma-international.org/ecma-262/5.1/#sec-C
> A conforming implementation, when processing strict mode code, may not extend the syntax of NumericLiteral (7.8.3) to include OctalIntegerLiteral as described in B.1.1.

If strict mode is specified, I think that they expect to find problem that doesn't follow the ecma262 strict mode spec.

Not only language version but also strict mode can be specified from the outside.
https://github.com/mozilla/rhino/blob/cee2661d816b1ba03d9174ddadcd65cc0d05bfcc/toolsrc/org/mozilla/javascript/tools/shell/Main.java#L325-L326

I think you should have done this if compatibility was taken into consideration.
```java
} else if (directive.equals("use strict")) {
    inUseStrictDirective = compilerEnv.getLanguageVersion() >= Context.VERSION_ES5;
```

Reply:
I'm not sure I understand. The goal is to maintain compatibility with the very large amount of legacy code that uses Rhino, and we have no idea how much of that code depends on "old octals." Ensuring that this change only affects versions >= VERSION_ES6 is the only way that we have right now to allow people interested in spec compliance to get it without breaking Rhino for everyone else. We've done this for a number of changes in the last few releases.

Even when strict mode is enabled, there is still code out there that will break if we make older versions incompatible.

Reply:
Do you want to revert it only when it is property name? It was an error from before when it is a expression.
```js
'use strict';
01; // SyntaxError from before
```
```js
'use strict';
({ 01: 1 }); // SyntaxError, if this pull request merged
```

Reply:
In general, I would like to avoid breaking that worked with the language version < VERSION_ES6 even if the older-version stuff was inconsistent like you described above. So if we can change this so that the change only affects code where the language version is >= VERSION_ES6 then I'm happy to merge it.

Reply:
Added `compilerEnv.getLanguageVersion() >= Context.VERSION_ES6`.